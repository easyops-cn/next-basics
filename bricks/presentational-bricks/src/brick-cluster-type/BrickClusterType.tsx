import React from "react";

import {
  BrickValueMapping,
  Color,
  MappingValue
} from "../brick-value-mapping/BrickValueMapping";

const mapping: { [key: string]: MappingValue } = {
  "-1": {
    text: "无",
    color: Color.gray
  },
  "0": {
    text: "开发",
    color: Color.purple
  },
  "1": {
    text: "测试",
    color: Color.green
  },
  "2": {
    text: "生产",
    color: Color.blue
  },
  "3": {
    text: "预发布",
    color: Color.cyan
  }
};

interface BrickClusterTypeProps {
  objectId?: string;
  value: "-1" | "0" | "1" | "2" | "3";
  showBg?: boolean;
}

export function BrickClusterType(
  props: BrickClusterTypeProps
): React.ReactElement {
  if (!props.objectId || props.objectId === "CLUSTER") {
    return (
      <BrickValueMapping
        value={props.value}
        mapping={mapping}
        showBg={props.showBg}
      />
    );
  } else {
    return <div>{props.value}</div>;
  }
}
