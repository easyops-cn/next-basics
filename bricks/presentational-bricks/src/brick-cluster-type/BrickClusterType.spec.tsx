import React from "react";
import { shallow } from "enzyme";
import { BrickClusterType } from "./BrickClusterType";

describe("BrickClusterType", () => {
  it("should work", () => {
    const wrapper = shallow(<BrickClusterType value="0" />);
    expect(wrapper).toMatchSnapshot();
  });
});
