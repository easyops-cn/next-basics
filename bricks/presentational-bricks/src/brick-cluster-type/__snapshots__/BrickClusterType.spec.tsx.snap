// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickClusterType should work 1`] = `
<BrickValueMapping
  mapping={
    Object {
      "-1": Object {
        "color": "gray",
        "text": "无",
      },
      "0": Object {
        "color": "purple",
        "text": "开发",
      },
      "1": Object {
        "color": "green",
        "text": "测试",
      },
      "2": Object {
        "color": "blue",
        "text": "生产",
      },
      "3": Object {
        "color": "cyan",
        "text": "预发布",
      },
    }
  }
  value="0"
/>
`;
